scmrepo/git/backend/dulwich/__init__.py

```diff
@@ -344,8 +344,12 @@ def is_tracked(self, path: str) -> bool:
         return False
 
     def is_dirty(self, untracked_files: bool = False) -> bool:
-        staged, unstaged, untracked = self.status()
-        return bool(staged or unstaged or (untracked_files and untracked))
+        if untracked_files:
+            staged, unstaged, untracked = self.status()
+            return bool(staged or unstaged or untracked)
+
+        staged, unstaged, _ = self.status(untracked_files="no")
+        return bool(staged or unstaged)
```

Comment:
We don't really promise backward compatibility at this point, so maybe consider adjusting the `is_dirty` flag